### PR TITLE
fix(chromium): associate navigation requests with navigations

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/frames.ts [hunk 1]
```diff
@@ -127,20 +127,30 @@ export class FrameManager {
...
+      return;
+    frame._pendingDocumentId = documentId;
   }
 
   frameCommittedNewDocumentNavigation(frameId: string, url: string, name: string, documentId: string, initial: boolean) {
     const frame = this._frames.get(frameId)!;
     this.removeChildFramesRecursively(frame);
     frame._url = url;
     frame._name = name;
+    // assert(!frame._pendingDocumentId || frame._pendingDocumentId === documentId);
```
Comment (dgozman) [inline src/frames.ts:151]: This does not hold? In which test?

File: test/autowaiting.spec.js [hunk 2]
```diff
@@ -196,7 +196,7 @@ describe('Auto waiting should not hang when', () => {
         setTimeout(() => window.stop(), 100);
       }, server.EMPTY_PAGE);
   });
-  it.fail(CHROMIUM)('calling window.stop sync', async({page, server, httpsServer}) => {
+  it('calling window.stop sync', async({page, server, httpsServer}) => {
     // Flaky, see https://github.com/microsoft/playwright/pull/1630/checks?check_run_id=553475173.
```
Comment (dgozman) [inline test/autowaiting.spec.js:200]: nit: please remove this comment. Also, reading the comment, I am not sure how does this patch address the issue :)

File: test/launcher.spec.js [hunk 3]
```diff
@@ -316,8 +316,7 @@ describe('browserType.launchPersistentContext', function() {
     await removeUserDataDir(userDataDir);
     await removeUserDataDir(userDataDir2);
   });
-  // See https://github.com/microsoft/playwright/issues/717
-  it.slow().fail(WIN && CHROMIUM)('userDataDir option should restore cookies', async({browserType, server}) => {
+  it.slow()('userDataDir option should restore cookies', async({browserType, server}) => {
```
Comment (dgozman) [inline test/launcher.spec.js:319]: Unrelated?

File: src/chromium/crNetworkManager.ts [hunk 4]
```diff
@@ -182,6 +182,8 @@ export class CRNetworkManager {
     }
     const isNavigationRequest = event.requestId === event.loaderId && event.type === 'Document';
     const documentId = isNavigationRequest ? event.loaderId : undefined;
+    if (isNavigationRequest)
+      this._page._frameManager.frameUpdatedDocumentIdForNavigation(event.frameId!, documentId!);
```
Comment (dgozman) [inline src/chromium/crNetworkManager.ts:186]: nit: might as well use `frame._id`

File: src/frames.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/frames.ts:151]: Fixed

File: test/autowaiting.spec.js (hunk 2 above)
Comment (pavelfeldman) [reply test/autowaiting.spec.js:200]: Done

File: src/chromium/crNetworkManager.ts (hunk 4 above)
Comment (pavelfeldman) [reply src/chromium/crNetworkManager.ts:186]: Nah

File: test/launcher.spec.js (hunk 3 above)
Comment (pavelfeldman) [reply test/launcher.spec.js:319]: It passes!

File: src/chromium/crConnection.ts [hunk 5]
```diff
@@ -163,7 +166,7 @@ export class CRSession extends EventEmitter {
         callback.resolve(object.result);
     } else {
       assert(!object.id);
-      Promise.resolve().then(() => this.emit(object.method!, object.params));
+      Promise.resolve().then(() => this.emit(object.method!, object.params)).catch(e => this._unhandledException = this._unhandledException || e);
```
Comment (dgozman) [inline src/chromium/crConnection.ts:169]: We should do the same for ff/wk.

**Review by dgozman — APPROVED** [review]

File: src/chromium/crConnection.ts (hunk 5 above)
Comment (pavelfeldman) [reply src/chromium/crConnection.ts:169]: Will follow up!
